internal/common/common.go

```diff
@@ -150,16 +150,13 @@ func GetBootstrapMachineNetworkAndIp(cluster *Cluster) (string, string) {
 		return "", ""
 	}
 	for _, intf := range inventory.Interfaces {
-		for _, addr := range intf.IPV4Addresses {
+		addresses := append(intf.IPV4Addresses, intf.IPV6Addresses...)
+		for _, addr := range addresses {
```

Comment:
You may parse the machine-cidr and then for each ip parse it and you can check parsed-machine-cidr.Contains(ip)